### DISCUSSION TITLE
Disable auto-suggestion/auto-complete on blink-cmp

### DISCUSSION THREAD

**Discussion by Squashedapricot** [LazyVim/LazyVim#5186] [Q&A] [answered]
Hi lads I am not able to find any documentation to disable the auto-complete/auto-suggestions on blink-cmp after the recent update, any ideas ?
Thanks for your time and Thank you folke and great contributors behind LazyVim, everyone here appreciates your hard work.

PS :- I have enable nvim-cmp temporarily which gets the job done for now.

**Comment by dpetka2001** [comment]
```lua
-- When true, will show the completion window after typing a character that matches the `keyword.regex`
show_on_keyword = false,
-- When true, will show the completion window after typing a trigger character
show_on_trigger_character = false,
```
should do what you want I believe. For more info, you should check out the blink docs to see what other options are available that might suit your needs more.

**Reply by Squashedapricot** [reply]
went through the readme here:- https://github.com/Saghen/blink.cmp 
didn't find anything mate (guess I'm blind). Can you link me the docs.

`return {
  "hrsh7th/nvim-cmp",
  opts = {
    completion = {
      keyword_length = 0,
      autocomplete = false,
    },
  },
}`

this is my current **nvim-cmp** disable setup
If it is possible can you give me **snippet like this** it **would mean the world to me**.
Again, **thanks for taking the time to reply**

**Reply by dpetka2001** [reply]
Try the following
```lua
return {
	{
		"saghen/blink.cmp",
		opts = {
			-- When true, will show the completion window after typing a character that matches the `keyword.regex`
			show_on_keyword = false,
			-- When true, will show the completion window after typing a trigger character
			show_on_trigger_character = false,
		},
	},
}
```
and see if that works for you.

**Reply by Squashedapricot** [reply]
Unfortunately, It did not work  :( 
By the way let me clarify, **I put the above config in lua/plugins/disabled.lua**
But **judging by the above comment I believe you know this already**

**Reply by dpetka2001** [reply]
Oh, you're probably on blink 0.7.6 stable release. I use blink main branch and those options work for me. Better wait until blink makes a new release, so the options will be available.

**Reply by Squashedapricot** [reply]
Yep, **you are right** just checked.
Thanks for you time man appreciate it.

**Comment by bytehaufen** [comment, marked as answer]
Why do not just use the `enabled` property?
I made it toggleable (and disabled by default) with

```lua
return {
  "saghen/blink.cmp",
  -- Make blink.cmp toogleable
  opts = function(_, opts)
    vim.b.completion = false

    Snacks.toggle({
      name = "Completion",
      get = function()
        return vim.b.completion
      end,
      set = function(state)
        vim.b.completion = state
      end,
    }):map("<leader>uk")

    opts.enabled = function()
      return vim.b.completion ~= false
    end
    return opts
  end,
}
```

**Reply by b11e** [reply]
this works for me,

**Reply by b11e** [reply]
although i had to change the enabled field to false, seems you have it set to true as it reads currently

**Reply by Squashedapricot** [reply]
**Exquisite, works for me Sir**. At least the staying disabled bit. 
Thank for your time Mr bytehaufen
**Will be marking your comment as the answer**

**Reply by AP0LL0-19** [reply]
Has anyone been able to remove the autocomplete from grep?

**Reply by folke** [reply]
autocomplete is by default disabled for prompt buffers, so if it shows up, it means **YOU** changed the default `enabled` setting for blink.cmp

**Reply by AP0LL0-19** [reply]
I doubt it as this is the only thing I have added to my blink-cmp.lua file in the plugins folder:

return {
  "saghen/blink.cmp",
  -- Make blink.cmp toogleable
  opts = function(_, opts)
    vim.b.completion = false

    Snacks.toggle({
      name = "Completion",
      get = function()
        return vim.b.completion
      end,
      set = function(state)
        vim.b.completion = state
      end,
    }):map("<leader>t")

    opts.enabled = function()
      return vim.b.completion ~= false
    end
    return opts
  end,

  keymaps = {
    preset = "default", --default accept is ctrl y
    ["<CR>"] = {}, --unbind enter as auto completion because it is annoying
  },
}

Also I've only noticed it with live grep

**Reply by dpetka2001** [reply]
But you **did change it**
```lua
opts.enabled = function()
  return vim.b.completion ~= false
end
```
That's what Folke is talking about. The default value can be seen [here](https://github.com/Saghen/blink.cmp/blob/b2485c76cb7877de6fe9c8670af59ba3d72fd74d/lua/blink/cmp/config/init.lua#L14).

**Reply by AP0LL0-19** [reply]
oh so should I remove that part. I just want to be able to toggle autocomplete on and off?

**Reply by dpetka2001** [reply]
That part doesn't really do anything with regards to your toggle. My suggestion would be to remove it.

**Reply by AP0LL0-19** [reply]
thank you, would you also be able to help with unbinding enter from accepting the autocomplete option?

**Reply by dpetka2001** [reply]
You can see blink's keymaps in [blink's docs](https://cmp.saghen.dev/configuration/keymap.html#presets). You can choose one preset that you like better or you can redefine all mappings to your liking. For disabling some mapping, you just set it to an empty table as also mentioned in the example in blink's docs.

**Reply by AP0LL0-19** [reply]
I tried that here: 
 keymaps = {
    preset = "default", --default accept is ctrl y
    ["<CR>"] = {}, --unbind enter as auto completion because it is annoying
  },

but it seems to have no affect.
you can see exactly where I do this change in my overall code in my previous post containing code

**Reply by dpetka2001** [reply]
It is `opts.keymap` not `opts.keymaps`.

**Reply by dpetka2001** [reply]
Oh just saw your previous comment more carefully. That was wrong. It has to be like this
```lua
return {
	{
		"saghen/blink.cmp",
		opts = {
			keymap = {
				preset = "default",
			},
		},
	},
}
```
or if you want to still keep `opts` a function like this
```lua
{
    "saghen/blink.cmp",
    opts = function(_, opts)
      opts.keymap = {
        preset = "default",
      }
    end,
  },
```

**Reply by tidux** [reply]
Can we make this toggleable-off-by-default the standard in LazyVim?  It's a great feature when programming, but people use (n)vim for a lot of other things besides that, and autocomplete/suggestions get distracting when trying to, say, write an email.  I've added it to my own config, I'm just thinking UX for new users.

**Comment by AP0LL0-19** [comment]
wow thanks for catching my mistakes, it all works good now? also when I post questions how can I get my code to look like the code you post, neatly highlighted instead of just text.

**Reply by dpetka2001** [reply]
You just write triple backticks followed by the language you want, then in a new line the code snippet and in another line of its own another set of triple backticks to close the code snippet.

**Reply by Jacobus-afk** [reply]
Just to add @AP0LL0-19, there's also a `Preview` button at the top of the writing text box, that you can use to see how your post is going to display

**Comment by Lanjelin** [comment]
Furthered configured this to allow the [list](https://cmp.saghen.dev/configuration/completion.html#list) to appear as a suggestion only, but having to manually select them -- with the option do disable it all together using the above `<leader>uk`.  

```lua
return {
  "saghen/blink.cmp",
  -- Make blink.cmp toogleable
  opts = function(_, opts)
    vim.b.completion = true

    Snacks.toggle({
      name = "Completion",
      get = function()
        return vim.b.completion
      end,
      set = function(state)
        vim.b.completion = state
      end,
    }):map("<leader>uk")

    opts.enabled = function()
      return vim.b.completion ~= false
    end

    -- Disable preselect and autocomplete
    opts.completion = opts.completion or {}
    opts.completion.list = opts.completion.list or {}
    opts.completion.list.selection = {
      preselect = false, -- Do not preselect
      auto_insert = false, -- Do not auto insert
    }
    return opts
  end,
}
```

**Reply by youlaibb** [reply]
Thank you so much! I was looking for this solution everywhere. I wish it would be the default setting in LazyVim.

**Reply by DuendeInexistente** [reply]
This doesn't work consistently for me, it works on the first tab but not subsequent ones. What's up with that?
